src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,116 @@
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows == 0) {
+                throw new SQLException("Creating book failed, no rows affected.");
+            }
+
+            try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    book.setId(generatedKeys.getLong(1));
+                } else {
+                    throw new SQLException("Creating book failed, no ID obtained.");
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create book " + book, e);
+        }
+
+        return book;
+    }
+
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql)) {
+            statement.setLong(1, id);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    Book book = new Book();
+                    book.setId(resultSet.getObject("id", Long.class));
+                    book.setPrice(resultSet.getObject("price", BigDecimal.class));
+                    book.setTitle(resultSet.getString("title"));
+                    return Optional.of(book);
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find by id for id = " + id, e);
+        }
+
+        return Optional.empty();
+    }
+
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT * FROM books";
+
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql);
+                ResultSet resultSet = statement.executeQuery()) {
+
+            while (resultSet.next()) {
+                Book book = new Book();
+                book.setId(resultSet.getObject("id", Long.class));
+                book.setPrice(resultSet.getObject("price", BigDecimal.class));
+                book.setTitle(resultSet.getString("title"));
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't retrieve all books", e);
+        }
+
+        return books;
+    }
+
+    public Book update(Book book) {
+        String sql = "UPDATE books SET price = ?, title = ? WHERE id = ?";
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql)) {
+            statement.setBigDecimal(1, book.getPrice());
+            statement.setString(2, book.getTitle());
+            statement.setLong(3, book.getId());
+            if (statement.executeUpdate() > 0) {
+                return book;
+            } else {
+                throw new DataProcessingException("No book found with id = " + book.getId(),
```

Comment:
Avoid throwing a generic `RuntimeException`. Instead, use the `SQLException` that is caught in the catch block to maintain the stack trace and cause of the original exception.